core/src/main/scala/com/github/johnynek/paiges/Doc.scala

```diff
@@ -293,16 +366,47 @@ object Doc {
     }
   }
 
-  private case object Empty extends Doc
-  private case class Concat(a: Doc, b: Doc) extends Doc
-  private case class Nest(indent: Int, doc: Doc) extends Doc
-  private case class Text(str: String) extends Doc
-  private case object Line extends Doc
   /**
-   * There is an additional invariant on Union that
-   * a == flatten(b). By construction all have this
-   * property, but this is why we don't expose Union
-   * but only .group
+   * Used internally by `Doc#compare`.
    */
-  private case class Union(a: Doc, b: Doc) extends Doc
+  sealed abstract class Tok
```

Comment:
eventually, I'd like these to be private, but I know we are still polishing this.